### PR TITLE
Add GIT_CLONE_PROTECTION_ACTIVE env config to fix build script

### PR DESCRIPTION
**Pull request by jadu-nv**
## Description

Closes https://github.com/nv-morpheus/Morpheus/issues/1742

This PR addresses the above Github Issue by configuring the `GIT_CLONE_PROTECTION_ACTIVE` environment variable in `docker/Dockerfile`

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -92,7 +92,7 @@ morpheus_utils_initialize_cuda_arch(morpheus)
 # Note intentionally excluding CUDA from the LANGUAGES list allowing us to set some clang specific settings later when
 # we call morpheus_utils_enable_cuda()
 project(morpheus
-  VERSION 24.06.00
+  VERSION 24.06.00rc1
```
Comment (dagardner-nv) [inline CMakeLists.txt:95]: This shouldn't be needed, and will need to change once we cut the release

File: examples/digital_fingerprinting/production/Dockerfile [hunk 2]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 
 ARG MORPHEUS_CONTAINER=nvcr.io/nvidia/morpheus/morpheus
-ARG MORPHEUS_CONTAINER_VERSION=v24.06.00-runtime
+ARG MORPHEUS_CONTAINER_VERSION=v24.06.00rc1-runtime
```
Comment (dagardner-nv) [inline examples/digital_fingerprinting/production/Dockerfile:17]: Same, shouldn't be needed

File: examples/digital_fingerprinting/production/docker-compose.yml [hunk 3]
```diff
@@ -109,7 +109,7 @@ services:
       target: runtime
       args:
         - MORPHEUS_CONTAINER=${MORPHEUS_CONTAINER:-nvcr.io/nvidia/morpheus/morpheus}
-        - MORPHEUS_CONTAINER_VERSION=${MORPHEUS_CONTAINER_VERSION:-v24.06.00-runtime}
+        - MORPHEUS_CONTAINER_VERSION=${MORPHEUS_CONTAINER_VERSION:-v24.06.00rc1-runtime}
```
Comment (dagardner-nv) [inline examples/digital_fingerprinting/production/docker-compose.yml:112]: These shouldn't be needed, if you want to build the DFP container with a custom version use the `MORPHEUS_CONTAINER_VERSION` environment variable.

**Review by dagardner-nv — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (jadu-nv) [reply CMakeLists.txt]: Reverted to `24.06.00`

File: examples/digital_fingerprinting/production/Dockerfile (hunk 2 above)
Comment (jadu-nv) [reply examples/digital_fingerprinting/production/Dockerfile]: Reverted to `24.06.00`

File: examples/digital_fingerprinting/production/docker-compose.yml (hunk 3 above)
Comment (jadu-nv) [reply examples/digital_fingerprinting/production/docker-compose.yml]: Reverted to `24.06.00`

**Review by dagardner-nv — APPROVED** [review]

**Pull request merged**